### ISSUE TITLE
Debugger halts with vscode_backtrack: Internal Error

### ISSUE THREAD

**Issue by dotnetCarpenter** [Microsoft/vscode#16059]
I'm trying to find out why `this.direction` is set to `" "`and is using a conditional breakpoint in order to inspect the stacktrace when `" "`is set.

But the debugger halts with *No call stack available (vscode_backtrack: Internal Error: TypeError: Cannot read property '0' of undefined).*

- VSCode Version: Code 1.7.2 (7ba55c5860b152d999dda59393ca3ebeb1b5c85f, 2016-11-21T22:14:18.217Z)
- OS Version: Linux x64 4.4.0-47-generic
- Extensions:

|Extension|Author|Version|
|---|---|---|
|beautify|HookyQR|0.5.5|
|ejs-language-support|QassimFarid|0.0.1|
|vscode-great-icons|emmanuelbeziat|1.1.25|
|vscode-apache|mrmlnc|1.1.0|
|spellchecker|swyphcosmo|1.2.3|
|ayu|teabyii|0.2.1|

---

Steps to Reproduce:

1. Clone https://github.com/dotnetCarpenter/eloquent.git
2. Checkout 4b8664c3b21db4da79bba12f748bf033685d5be1
3. Add conditional breakpoint on [line 116 in lifelikeSimulation.js](https://github.com/dotnetCarpenter/eloquent/commit/4b8664c3b21db4da79bba12f748bf033685d5be1#diff-5b51c790755f118bdc69ac07110b1b6eR116)
4. The condition should be `value == " "`
5. Start the debugger with the following launch.json:
```json
{
	"version": "0.2.0",
	"configurations": [{
		"type": "node",
		"request": "launch",
		"name": "Launch Lifelike",
		"program": "${workspaceRoot}/ch07/lifelike.js",
		"cwd": "${workspaceRoot}/ch07"
	}]
}
```

**Comment by dotnetCarpenter** [comment]
If you have a similar issue, a work-around is to add a condition to your source code. In the above case, it could be `if(value == " ") debugger`

**Comment by weinand** [comment]
@dotnetCarpenter please try our new experimental node debugger by setting the attribute `type` in your launch config to `node2` (instead of `node`).

**Comment by weinand** [comment]
@dotnetCarpenter what version of node are you using?

**Comment by dotnetCarpenter** [comment]
@weinand node: v7.2.0
Thanks. Using `node2` works for *launch* but not for *attach* (nothing happens). `node` works for *attach* but fails as described in the first post.

**Comment by dotnetCarpenter** [comment]
When I use *attach*, I start the process with `node --debug-brk lifelike.js`

**Comment by weinand** [comment]
@dotnetCarpenter running lifelike.js on macOS with node v7.2 never results in  `value == " "` becoming true. That makes it difficult to reproduce.

**Comment by dotnetCarpenter** [comment]
@weinand  Did you checkout commit 4b8664c3b21db4da79bba12f748bf033685d5be1 ?

**Comment by dotnetCarpenter** [comment]
I found and fix my bug but if (step 2) you checkout `4b8664c3b21db4da79bba12f748bf033685d5be1` then you should hit the bug which is due to [line 156 in ch07/lifelikeSimulation.js](https://github.com/dotnetCarpenter/eloquent/blob/4b8664c3b21db4da79bba12f748bf033685d5be1/ch07/lifelikeSimulation.js#L156)

```js
findPlant(view) {
	// look for plant around
	const plant = view.find("*")
	if(plant) {
		this.direction = plant
		return { type: "eat", direction: plant }
	}

	// if not then look for space in the direction we are heading
	this.direction = view.look(this.direction)// <- this line courses this.direction = " "
	if(this.direction != " ")
		this.direction = view.find(" ") || "s"

	return { type: "move", direction: this.direction }
	}
}
```

**Comment by weinand** [comment]
this has been fixed in March 2017
